Automattic/WP-Job-Manager
# Add redirect to promote job form  on WPJMCOM

Part of #2439 

### Changes proposed in this Pull Request

- Make `get_api_base_url` public, so we can use it elsewhere in the code to get a dynamic link do WPJMCOM
- Make Promote Job button work, redirecting to WPJMCOM (whether it is a dev version or not) with the appropriate parameters
- Add endpoint to generate token dynamically and redirect the user appropriately
- Refactor the action to deactivate a promotion to use the `admin_action_` hook.

### Testing instructions

Using this branch of the plugin:

1. Register a job on WPJM
2. On the admin page listing all the jobs (`/wp-admin/edit.php?post_type=job_listing`), click on the "Promote" button
3. On the modal that appears, click on the "Promote your jobs" button
4. Check that the plugin validates that you're using HTTPS (if you're not, feel free to comment [this line for testing](https://github.com/Automattic/WP-Job-Manager/pull/2481/files#diff-48ee1884d5a044341d0f6de2f55553f95039416b06b775a572afb6ea907a10a3R90))
5. Make sure you're directed to a page like `https://wpjobmanager.com/promote-job?user_id=YOUR_USER_ID&job_id=YOUR_JOB_ID&token=A_RANDOM_TOKEN&site_url=THE_HTTPS_VERSION_OF_YOUR_SITE`

Also, try setting your WPJM version to include the `-dev` suffix, and add the constant `JOB_MANAGER_DEV_API_BASE_URL` in your `wp-config.php` setting the URL for your local WPJMCOM instance. Make sure the above link is changed appropriately. 

Finally, check out if the option to deactivate a promotion still works, by setting a job listing with `_promoted` set to `1` and verifying if the deactivate flow still works.
